src/sst/core/params.cc

```diff
@@ -480,6 +480,54 @@ Params::getArrayTokens(const std::string& value, std::vector<std::string>& token
     }
 }
 
+
+std::map<std::string, std::string>
+Params::getGlobalParamSet(const std::string& name)
+{
+    std::map<std::string, std::string> ret;
+    auto                               it = global_params.find(name);
+    if ( it == global_params.end() ) return ret;
+
+    for ( auto x : it->second ) {
+        ret[getParamName(x.first)] = x.second;
+    }
+    return ret;
+}
+
+std::vector<std::string>
+Params::getGlobalParamSetNames()
+{
+    std::vector<std::string> ret;
```

Comment:
If we know the size of the global params we should reserve

Reply:
Fixed